### ISSUE TITLE
Inspections for the Rust Style Guide

### ISSUE THREAD

**Issue by johnthagen** [intellij-rust/intellij-rust#431]
Would be great if `intellij-rust` came out of the box configured to enforce the [rust style guide](https://github.com/rust-lang/rust/tree/master/src/doc/style), similar to how PyCharm is configured out of the box for [PEP8](https://www.python.org/dev/peps/pep-0008/).

**Comment by matklad** [comment]
Yeah, the main problem with this is that rust style guide is not currently formalized. There is a plan to establish RFC-like process to decide on the Rust style. 

See this discussion for some more details: https://users.rust-lang.org/t/please-help-test-rustfmt/5386

**Comment by matklad** [comment]
Relevant RFC: https://github.com/rust-lang/rfcs/pull/1607

**Comment by johnthagen** [comment]
For reference, a Rust style team has been formed and is working out of this repo: https://github.com/rust-lang-nursery/fmt-rfcs

**Comment by matklad** [comment]
The style guide is not finalized yet, but it makes sense to go through parts that are more or less finished and make sure our formatter does the same. 

Formatter is here: https://github.com/intellij-rust/intellij-rust/tree/master/src/main/kotlin/org/rust/ide/formatter.

CC #1129

**Comment by Kobzol** [comment]
@mchernyavsky I suppose that this can be now closed, since it's now possible to use `rustfmt` by default?
